### PR TITLE
refactor: update all collaboration copy to connection

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR updates all `collaboration` copy to `connection`

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

### Before 
Screenshot 2023-10-09 at 17 26 46

### After
Screenshot 2023-10-09 at 17 22 42

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by OgDev-01** [comment]
Opening a PR on the API side to reflect same changes shortly to unblock this

**Comment by OgDev-01** [comment]
once https://github.com/open-sauced/api/pull/354 is landed, this will be ready for review... converting to draft

File: components/organisms/ConnectionRequestWrapper/connection-requests-wrapper.tsx [hunk 1]
```diff
@@ -60,13 +60,13 @@ const CollaborationRequestsWrapper = () => {
         <DashContainer className="flex-col gap-6 md:gap-8 text-light-slate-9">
           <p>No pending requests.</p>
           <p className="text-center md:px-16">
-            Sometimes you got to be a friend to make a friend. Considering requesting collaboration on sharing profile
-            on social channels like GitHub or Twitter
+            Sometimes you got to be a friend to make a friend. Considering requesting connections on sharing profile on
```
Comment (bdougie) [inline components/organisms/ConnectionRequestWrapper/connection-requests-wrapper.tsx:63]:
```suggestion
            Sometimes you got to be a friend to make a friend. Considering inviting other developers to OpenSauced.
```

File: components/organisms/ConnectionRequestWrapper/connection-requests-wrapper.tsx [hunk 2]
```diff
@@ -60,13 +60,13 @@ const CollaborationRequestsWrapper = () => {
...
+            social channels like GitHub or Twitter
```
Comment (bdougie) [inline components/organisms/ConnectionRequestWrapper/connection-requests-wrapper.tsx:64]:
```suggestion
```

Does this need to be two lines?

**Review by bdougie — CHANGES_REQUESTED** [review]
Some copy doesn't make sense

File: lib/hooks/updateEmailPreference.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { supabase } from "lib/utils/supabase";
 
 interface UpdateEmailProps {
   display_email?: boolean;
-  receive_collaboration?: boolean;
+  receive_connections?: boolean;
```
Comment (brandonroberts) [inline lib/hooks/updateEmailPreference.ts:5]: This is going to fail when it hits the API

File: lib/hooks/useUserConnections.ts [hunk 4]
```diff
@@ -3,24 +3,24 @@ import publicApiFetcher from "lib/utils/public-api-fetcher";
...
   const { toast } = useToast();
-  const { data, error, mutate } = useSWR<UserCollaborationResponse, Error>(
+  const { data, error, mutate } = useSWR<UserConnectionResponse, Error>(
     "user/collaborations",
-    publicApiFetcher as Fetcher<UserCollaborationResponse, Error>
+    publicApiFetcher as Fetcher<UserConnectionResponse, Error>
   );
 
-  async function requestCollaboration(payload: collaborationRequestPayload) {
+  async function requestConnection(payload: connectionsRequestPayload) {
```
Comment (brandonroberts) [inline lib/hooks/useUserConnections.ts:23]: Same here. The request sent to the API should remain the same

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: lib/hooks/updateEmailPreference.ts (hunk 3 above)
Comment (OgDev-01) [reply lib/hooks/updateEmailPreference.ts:5]: missed this 🥹

File: lib/hooks/useUserConnections.ts (hunk 4 above)
Comment (OgDev-01) [reply lib/hooks/useUserConnections.ts:23]: The request are still the same here

File: components/organisms/UserSettingsPage/user-settings-page.tsx [hunk 5]
```diff
@@ -359,11 +359,11 @@ const UserSettingsPage = ({ user }: userSettingsPageProps) => {
               <Checkbox
                 onCheckedChange={() =>
                   // eslint-disable-next-line camelcase
-                  setEmailPreference((prev) => ({ ...prev, receive_collaboration: !prev.receive_collaboration }))
+                  setEmailPreference((prev) => ({ ...prev, receive_connections: !prev.receive_collaboration }))
```
Comment (brandonroberts) [inline components/organisms/UserSettingsPage/user-settings-page.tsx:362]:
```suggestion
                  setEmailPreference((prev) => ({ ...prev, receive_collaboration: !prev.receive_collaboration }))
```

**Review by bdougie — APPROVED** [review]

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: next-types.d.ts [hunk 6]
```diff
@@ -313,7 +313,7 @@ interface DbUserCollaboration {
   readonly updated_at: string;
   readonly deleted_at: string;
   readonly request_emailed_at: string;
-  readonly collaboration_emailed_at: string;
+  readonly collaborations_emailed_at: string;
```
Comment (brandonroberts) [inline next-types.d.ts:316]:
```suggestion
  readonly collaboration_emailed_at: string;
```

**Pull request merged**